@navikt/core/react/src/form/search-field/SearchField.tsx

```diff
@@ -99,34 +136,52 @@ const SearchField = forwardRef<HTMLDivElement, SearchFieldProps>(
           <BodyShort
             as="div"
             className={cl("navds-text-field__description", {
-              "sr-only": hideLabel,
+              "navds-sr-only": hideLabel,
             })}
             id={inputDescriptionId}
             size={size}
           >
             {description}
           </BodyShort>
         )}
-        <div className="navds-search-field__input-wrapper">
-          <SearchFieldContext.Provider
-            value={{
-              inputProps,
-              size,
-            }}
-          >
-            {children}
-          </SearchFieldContext.Provider>
-        </div>
-        <div id={errorId} aria-relevant="additions removals" aria-live="polite">
-          {showErrorMsg && <ErrorMessage size={size}>{error}</ErrorMessage>}
+        <div
+          data-value={!!controlledValue}
+          className="navds-search-field__input-wrapper"
+        >
+          <span className="navds-search-field__input-icon">
+            <Search aria-hidden />
+          </span>
+          <input
+            ref={mergedRef}
+            {...omit(rest, ["size"])}
+            {...inputProps}
+            {...(props.value !== undefined && { value: props.value })}
+            onChange={(e) => handleChange(e.target.value)}
+            type="search"
+            role="searchbox"
+            className={cl(
+              className,
+              "navds-search-field__input",
+              "navds-text-field__input",
+              "navds-body-short",
+              `navds-body-${size ?? "medium"}`
+            )}
+          />
+          {controlledValue && clearButton && (
+            <button
+              onClick={() => handleClear()}
+              className="navds-search-field__clear-button"
+            >
```

Comment:
Kjedelig at vi ikke får brukt `<Button>`-komponenten her. Kanskje det er verdt å ta en runde med designer for å høre om vi ikke kan gjenbruke det uttrykket?

Reply:
Jørgen var ganske klar på ingen border på hover, endring av ikon-farge på hover etc, men kan høre me han.

Vil være mulig å ta i bruk tertiary-button her, men usikker på om det er en "bedre" løsning å overskrive CSS til knappen når den er i context av søkefelt. Hvis vi har kunne brukt nesting av css-variabler ville dette vært mye enklere. Eks: 
```css
.navds-search-field_clear-button {
	  --navds-button-color-tertiary-border-hover: transparent;
	  --navds-button-color-tertiary-hover: var(--navds-semantic-color-interaction-primary);
}
```
Ønsker å ta opp diskusjonen om bruk av dette igjen da IE11 har End of life nå i Juni 2022.